demisto_sdk/commands/upload/new_uploader.py

```diff
@@ -0,0 +1,301 @@
+            if os.path.basename(entity_folder.rstrip('/')) in CONTENT_ENTITIES_DIRS:
+                status_code = self.entity_dir_uploader(entity_folder) or status_code
+        return status_code
+
+
+def parse_error_response(error: ApiException, file_type: str, file_name: str, print_error: bool = False):
+    """
+    Parses error message from exception raised in call to client to upload a file
+
+    error (ApiException): The exception which was raised in call in to client
+    file_type (str): The file type which was attempted to be uploaded
+    file_name (str): The file name which was attempted to be uploaded
+    """
+    message = error
+    if hasattr(error, 'reason'):
+        if '[SSL: CERTIFICATE_VERIFY_FAILED]' in str(error.reason):
+            message = '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate.\n' \
+                      'Try running the command with --insecure flag.'
+
+        elif 'Failed to establish a new connection:' in str(error.reason):
+            message = 'Failed to establish a new connection: Connection refused.\n' \
+                      'Try checking your BASE url configuration.'
+
+        elif error.reason in ('Bad Request', 'Forbidden'):
+            error_body = json.loads(error.body)
+            message = error_body.get('error')
+
+            if error_body.get('status') == 403:
+                message += '\nTry checking your API key configuration.'
+    if print_error:
+        click.secho(str(f'\nUpload {file_type}: {file_name} failed:'), fg='bright_red')
+        click.secho(str(message), fg='bright_red')
+    return message
+
+
+def print_summary(successfully_uploaded_files, unuploaded_due_to_version, failed_uploaded_files):
+    """Prints uploaded files summary
+    Successful uploads grid based on `successfully_uploaded_files` attribute in green color
+    Failed uploads grid based on `failed_uploaded_files` attribute in red color
+    """
+    click.secho('\n\nUPLOAD SUMMARY:')
+    if successfully_uploaded_files:
+        click.secho('\nSUCCESSFUL UPLOADS:', fg='green')
+        click.secho(tabulate(successfully_uploaded_files, headers=['NAME', 'TYPE'],
+                             tablefmt="fancy_grid") + '\n', fg='green')
+    if unuploaded_due_to_version:
+        click.secho('\nNOT UPLOADED DUE TO VERSION MISMATCH:', fg='yellow')
+        click.secho(tabulate(unuploaded_due_to_version, headers=['NAME', 'TYPE', 'XSOAR Version',
+                                                                 'FILE_FROM_VERSION', 'FILE_TO_VERSION'],
+                             tablefmt="fancy_grid") + '\n', fg='yellow')
+    if failed_uploaded_files:
+        click.secho('\nFAILED UPLOADS:', fg='bright_red')
+        click.secho(tabulate(failed_uploaded_files, headers=['NAME', 'TYPE', 'ERROR'],
+                             tablefmt="fancy_grid") + '\n', fg='bright_red')
+
+
+def sort_directories_based_on_dependencies(dir_list: List) -> List:
```

Comment:
This was in the old upload. I copied it. I believe it's for uploading non dependant enitites firs i.e. integration before playbook because the playbook uses the integration.